### ISSUE TITLE
Error: Null safety features are disabled for this library.

### ISSUE THREAD

**Issue by pedromassango** [jamesblasco/layout#13]
Hi,

I'm facing a strange error while trying to run my project with this package. It says that Null safety is disabled for this package.

I've tried to set my SDK version as the as this package but this does not work as well.

`layout: ^0.6.0`

<details>
<summary>logs</summary>

```bash
Launching lib/main.dart on Chrome in debug mode...
Waiting for connection from debug service on Chrome...
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/value.dart:39:5: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    required T xs,
    ^^^^^^^^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/value.dart:40:6: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    T? sm,
     ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/value.dart:41:6: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    T? md,
     ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/value.dart:42:6: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    T? lg,
     ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/value.dart:43:6: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    T? xl,
     ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/value.dart:108:10: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
  final T? sm;
         ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/value.dart:109:10: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
  final T? md;
         ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/value.dart:110:10: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
  final T? lg;
         ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/value.dart:111:10: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
  final T? xl;
         ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/value.dart:114:5: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    required this.xs,
    ^^^^^^^^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/value.dart:122:5: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    required T xs,
    ^^^^^^^^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/value.dart:123:5: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    required T sm,
    ^^^^^^^^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/value.dart:124:5: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    required T md,
    ^^^^^^^^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/value.dart:125:5: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    required T lg,
    ^^^^^^^^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/value.dart:126:5: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    required T xl,
    ^^^^^^^^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/value.dart:133:62: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
  BreakpointValue.fromMap(Map<LayoutBreakpoint, T> values, [T? defaultValue])
                                                             ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/margin.dart:8:19: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
  final EdgeInsets? margin;
                  ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/margin.dart:11:8: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    Key? key,
       ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/margin.dart:12:5: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    required this.child,
    ^^^^^^^^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/margin.dart:28:5: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    required LayoutData layout,
    ^^^^^^^^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/margin.dart:30:15: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    EdgeInsets? margin,
              ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/margin.dart:31:11: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    double? maxWidth,
          ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/margin.dart:48:19: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
  final EdgeInsets? margin;
                  ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/margin.dart:51:8: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    Key? key,
       ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/margin.dart:52:5: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    required Widget sliver,
    ^^^^^^^^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/margin.dart:57:8: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    Key? key,
       ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/margin.dart:58:5: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    required Widget sliver,
    ^^^^^^^^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/margin.dart:97:15: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
  final double? maxWidth;
              ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/margin.dart:98:19: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
  final EdgeInsets? margin;
                  ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/margin.dart:101:8: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    Key? key,
       ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/margin.dart:102:5: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    required this.child,
    ^^^^^^^^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/margin.dart:123:15: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
  final double? maxWidth;
              ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/margin.dart:124:19: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
  final EdgeInsets? margin;
                  ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/margin.dart:128:8: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    Key? key,
       ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/margin.dart:129:5: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    required Widget sliver,
    ^^^^^^^^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/layout.dart:28:8: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    Key? key,
       ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/layout.dart:29:5: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    required this.child,
    ^^^^^^^^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/layout.dart:35:21: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
  final LayoutFormat? format;
                    ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/layout.dart:41:20: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
  static LayoutData? maybeOf(BuildContext context) => context
                   ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/layout.dart:51:8: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    Key? key,
       ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/layout.dart:52:5: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    required this.size,
    ^^^^^^^^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/layout.dart:53:5: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    required this.margin,
    ^^^^^^^^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/layout.dart:54:5: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    required this.format,
    ^^^^^^^^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/layout.dart:55:5: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    required this.gutter,
    ^^^^^^^^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/layout.dart:56:5: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    required this.breakpoint,
    ^^^^^^^^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/layout.dart:57:5: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    required this.columns,
    ^^^^^^^^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/layout.dart:58:5: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    required this.maxWidth,
    ^^^^^^^^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/layout.dart:72:15: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
  T value<T>({required T xs, T? sm, T? md, T? lg, T? xl}) {
              ^^^^^^^^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/layout.dart:72:31: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
  T value<T>({required T xs, T? sm, T? md, T? lg, T? xl}) {
                              ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/layout.dart:72:38: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
  T value<T>({required T xs, T? sm, T? md, T? lg, T? xl}) {
                                     ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/layout.dart:72:45: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
  T value<T>({required T xs, T? sm, T? md, T? lg, T? xl}) {
                                            ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/layout.dart:72:52: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
  T value<T>({required T xs, T? sm, T? md, T? lg, T? xl}) {
                                                   ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/layout.dart:110:8: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    Key? key,
       ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/layout.dart:111:5: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    required this.data,
    ^^^^^^^^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/layout.dart:112:5: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    required Widget child,
    ^^^^^^^^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/breakpoint.dart:10:19: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
  LayoutBreakpoint? get smaller {
                  ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/breakpoint.dart:20:19: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
  LayoutBreakpoint? get bigger {
                  ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/gutter.dart:6:15: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
  final double? extent;
              ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/gutter.dart:9:8: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    Key? key,
       ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/gutter.dart:19:15: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
  final double? extent;
              ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/gutter.dart:22:8: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    Key? key,
       ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/formats/fluid_format.dart:10:24: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    LayoutValue<double>? margin,
                       ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/formats/material_format.dart:7:24: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    LayoutValue<double>? margin,
                       ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/formats/material_format.dart:8:24: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    LayoutValue<double>? gutter,
                       ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/formats/boostrap_format.dart:7:24: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    LayoutValue<double>? margin,
                       ^
Error: Cannot run with sound null safety, because the following dependencies
don't support null safety:

 - package:layout

For solutions, see https://dart.dev/go/unsound-null-safety
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/value.dart:138:62: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
        this.xs = values[LayoutBreakpoint.xs] ?? defaultValue!,
                                                             ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/layout.dart:38:68: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
      .dependOnInheritedWidgetOfExactType<_LayoutInheritedWidget>()!
                                                                   ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/formats/fluid_format.dart:20:65: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
            return min(containerWidth, maxFluidWidth[breakpoint]!);
                                                                ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/formats/fluid_format.dart:26:45: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
            return maxFluidWidth[breakpoint]!;
                                            ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/formats/fluid_format.dart:38:66: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
    final totalDistance = width - breakpoints[breakpoint.smaller]!;
                                                                 ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/formats/fluid_format.dart:41:34: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
        maxFluidWidth[breakpoint]! - maxFluidWidth[breakpoint.smaller]!;
                                 ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/formats/fluid_format.dart:41:71: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
        maxFluidWidth[breakpoint]! - maxFluidWidth[breakpoint.smaller]!;
                                                                      ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/formats/fluid_format.dart:44:41: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
        maxFluidWidth[breakpoint.bigger]! - totalFluidDistance * progress;
                                        ^
../../../Code/flutter_beta/.pub-cache/hosted/pub.dartlang.org/layout-0.6.0/lib/src/format.dart:65:43: Error: Null safety features are disabled for this library.
Try removing the package language version or setting the language version to 2.12 or higher.
          width >= breakpoints[breakpoint]!) {
                                          ^
Unhandled exception:
Bad state: Unsupported Null Safety mode NonNullableByDefaultCompiledMode.Invalid, in null.
#0      ProgramCompiler.emitModule (package:dev_compiler/src/kernel/compiler.dart:435:9)
#1      JavaScriptBundler.compile (package:frontend_server/src/javascript_bundle.dart:144:33)
#2      FrontendCompiler.writeJavascriptBundle (package:frontend_server/frontend_server.dart:632:20)
<asynchronous suspension>
#3      FrontendCompiler.compile (package:frontend_server/frontend_server.dart:545:9)
<asynchronous suspension>
#4      listenAndCompile.<anonymous closure> (package:frontend_server/frontend_server.dart:1105:11)
<asynchronous suspension>
Finished with error: the Dart compiler exited unexpectedly.
Failed to compile application.
```
</details>

**Comment by jamesblasco** [comment]
I am sorry about this issue @pedromassango. The SDK constraints are wrong for version '0.6.0'. 

The package is built with null safety, but this feature is not stable yet. My readme information was not displaying correctly in pub.dev. So I changed the reduce the min dart sdk version for that. 

I will publish a new version with the proper sdk constraints. But notice this package will be available beginning with dart  2.12

**Comment by jamesblasco** [comment]
This should be fixed in `0.6.1` 

https://pub.dev/packages/layout/versions/0.6.1

**Comment by pedromassango** [comment]
Great let me try it...

**Comment by deandreamatias** [comment]
@pedromassango if this issue was solved with version 0.6.1, please close the issue
